util/buildflags/export_test.go

```diff
@@ -0,0 +1,118 @@
+package buildflags
+
+import (
+	"cmp"
+	"slices"
+	"testing"
+
+	gocmp "github.com/google/go-cmp/cmp"
+	"github.com/moby/buildkit/exporter/containerimage/exptypes"
+	ocispecs "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestParseAnnotations(t *testing.T) {
+	tests := []struct {
+		name    string
+		in      []string
+		want    map[exptypes.AnnotationKey]string
+		wantErr string
+	}{
+		{
+			name: "basic",
+			in:   []string{"a=b"},
+			want: map[exptypes.AnnotationKey]string{
+				{Key: "a"}: "b",
+			},
+		},
+		{
+			name: "reverse-DNS key",
+			in:   []string{"com.example=a"},
+			want: map[exptypes.AnnotationKey]string{
+				{Key: "com.example"}: "a",
+			},
+		},
+		{
+			name: "specify type",
+			in:   []string{"manifest:com.example=a"},
+			want: map[exptypes.AnnotationKey]string{
+				{Type: "manifest", Key: "com.example"}: "a",
+			},
+		},
+		{
+			name:    "specify bad type",
+			in:      []string{"bad:com.example=a"},
+			wantErr: "unknown annotation type",
+		},
+		{
+			name: "specify type and platform",
+			in:   []string{"manifest[plat/form]:com.example=a"},
+			want: map[exptypes.AnnotationKey]string{
+				{
+					Type: "manifest",
+					Platform: &ocispecs.Platform{
+						OS:           "plat",
+						Architecture: "form",
+					},
+					Key: "com.example",
+				}: "a",
+			},
+		},
+		{
+			name: "specify multiple types",
+			in:   []string{"index,manifest:com.example=a"},
+			want: map[exptypes.AnnotationKey]string{
+				{Type: "index", Key: "com.example"}:    "a",
+				{Type: "manifest", Key: "com.example"}: "a",
+			},
+		},
+		{
+			name: "specify multiple types and platform",
+			in:   []string{"index,manifest[plat/form]:com.example=a"},
+			want: map[exptypes.AnnotationKey]string{
+				{Type: "index", Key: "com.example"}: "a",
+				{
+					Type: "manifest",
+					Platform: &ocispecs.Platform{
+						OS:           "plat",
+						Architecture: "form",
+					},
+					Key: "com.example",
+				}: "a",
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			got, gotErr := ParseAnnotations(test.in)
+			if test.wantErr != "" {
+				require.ErrorContains(t, gotErr, test.wantErr)
+			} else {
+				assert.NoError(t, gotErr)
+			}
+
+			// Can't compare maps with pointer in their keys, need to extract and sort the map entries
```

Comment:
might make sense to make this covert code into a function and then convert both maps. Atm there is duplication where exactly the same commands need to run on both maps.

Reply:
done